### ISSUE TITLE
Docker operations should use DOCKER_HOST env var when present

### ISSUE THREAD

**Issue by BrianAdams** [resin-io/resin-cli#625]
I am running inside a Dind container on a CI server.  The normal docker commands work just fine.  There is an environment setting that specifies the client should be connecting via a tcp connection:

DOCKER_HOST=tcp://docker:2375

Yet the resin client does not seem to honor that setting and keeps trying to connect to the /var/run/docker.sock (according to the error message).  IMHO, the Docker library in use should be configured to accept DOCKER_HOST and connect correctly to the running docker instance.

>resin preload $(pwd)/resin.img --api-key $RESIN_PROD_APIKEY --app $APP_ID --commit $COMMITHASH
```
Error: Docker seems to be unavailable (using socket /var/run/docker.sock). Is it installed, and do you have permission to talk to it?
    at /usr/lib/node_modules/resin-cli/build/utils/docker.js:330:13
    at tryCatcher (/usr/lib/node_modules/resin-cli/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/usr/lib/node_modules/resin-cli/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/usr/lib/node_modules/resin-cli/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/usr/lib/node_modules/resin-cli/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/usr/lib/node_modules/resin-cli/node_modules/bluebird/js/release/promise.js:689:18)
    at Async._drainQueue (/usr/lib/node_modules/resin-cli/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/usr/lib/node_modules/resin-cli/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/usr/lib/node_modules/resin-cli/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
```

**Comment by pimterry** [comment]
Thanks for the report @BrianAdams! We'll take a look at this, I think you're probably right that we should be honouring that.

In the meantime, you should be able to work around this by specifying the connection options explicitly:

```
resin preload --dockerHost docker --dockerPort 2375 [...other arguments]
```

Those same arguments should work for all commands that need to use a Docker instance (i.e. `resin build/deploy` too). Let me know if you have any trouble.

**Comment by BrianAdams** [comment]
That worked. Thanks. Recommend adding as options to the help documentation!

**Comment by pimterry** [comment]
This is fixed: `DOCKER_HOST` will now be observed automatically for all docker-based commands, if no explicit connection arguments are provided. This has been released in v6.6.2.
